### DISCUSSION TITLE
sisotool(sys) isn't ploting

### DISCUSSION THREAD

**Discussion by eletroinf** [python-control/python-control#770] [Q&A]
Hello
I'm trying python control, starting with the examples.
With this simple example:

from control.matlab import *
sys = tf([1000], [1,25,100,0])
sisotool(sys)

Located at https://python-control.readthedocs.io/en/latest/generated/control.sisotool.html
What I understand is that it should generate a plot, but when I run the code from a python terminal nothing happens.
Running it from Anaconda spyder it generates a figure plot, what I think it is the correct behavior.

It wouldn't generate a figure when running in a terminal? (I'm using windows)

Regards
Ricardo

**Comment by bnavigator** [comment]
Hi,

Did you set up your interactive terminal to plot Matplotlib figures?
https://matplotlib.org/stable/users/explain/interactive.html

**Reply by eletroinf** [reply]
Hi,
I'm trying to understand what you mean, but for me it doesn't make sense. pytplot isn't even imported in the example code. I think it is imported inside the control library. Documentation doesn't mention that it is needed to set pyplot.

Thank you anyway

**Comment by eletroinf** [comment]
Hi all,
Just to close this issue, today I worked on it. One solution to this code to work is import pyplot and use the function plt.show() at the end.
By doing this, it will show a picture (I'm using Python 3.11).

import matplotlib.pyplot as plt
from control.matlab import *

sys = tf([1000], [1,25,100,0])
sisotool(sys)


plt.show()
